version1/04 - Web services.md

```diff
@@ -1780,3 +1780,422 @@ Authorization: Bearer ACCESS_TOKEN
 **Response**
 
 204 No Content
+
+## Managing code list permissions
+
+This set of endpoints allows you to manage the explicit permissions on the specified objects:
+* all code lists
+* just the administrativEnhet code list or any of its list values
+* just the mappetype code list or any of its list values
+* just the dokumenttype code list or any of its list values
+* just the skjerming code list or any of its list values
+
+Address:
+```
+https://{server}:{port}/rms/api/public/noark5/v1/permission/code-list
+```
+
+### Get existing permissions
+
+Gets existing permissions for all code lists, a single code list, or a specific code list value.
+
+The use cases are, respectively:
+* GET /rms/api/public/noark5/v1/permission/code-list?listType=all
+* GET /rms/api/public/noark5/v1/permission/code-list?listType=administrativEnhet|mappetype|dokumenttype|skjerming
+* GET /rms/api/public/noark5/v1/permission/code-list?listType=administrativEnhet|mappetype|dokumenttype|skjerming&code=CODE
+
+**Request**
+
+``` text
+GET /rms/api/public/noark5/v1/permission/code-list?groupId=LONG&listType=STRING&code=STRING&offset=INTEGER&limit=INTEGER HTTP/1.1
+Authorization: Bearer ACCESS_TOKEN
+```
+
+###### Details
+
+- **groupId** (optional)
+  - the group ID for which to fetch existing permissions
+  - if omitted, all permissions for the specified object will be returned ordered by group identifier (asc)
+- **listType**
+  - the list type for which to fetch existing permissions
+  - could be one of 'all', 'admnistrativEnhet', 'mappetype', 'dokumenttype', 'skjerming', where 'all' represents permissions applicable to all code lists and the other values represent permissions applied on a specific code list
+  - 'all' can be considered as an entity of its own that is the parent of existing code lists
+- **code** (optional)
+  - the specific code list value for which to fetch existing permissions
+  - when present, only the permissions applied on the specified code list value will be returned
+- **offset** (optional)
+  - offset of first result
+  - defaults to 0
+  - must be greater than or equal to 0
+- **limit** (optional)
+  - maximum number of results to retrieve
+  - must be greater than 0 and less than or equal to 200
+  - defaults to 10
+
+**Response**
+
+200 OK
+
+```
+Content-type: application/json
+
+{
+    "permissions": [
+      {
+        "groupId": long,
+        "listType": string,
+        "code": string,
+        "explicitPermissions": [string]   
+      }
+    ],
+    "hasMore": boolean
+}
+```
+
+###### Details
+
+- **groupId**
+  - the group ID for which permissions were assigned
+- **listType**
+  - the list type for which permissions were assigned
+  - could be one of 'all', 'admnistrativEnhet', 'mappetype', 'dokumenttype', 'skjerming', where 'all' represents permissions applicable to all code lists and the other values represent permissions applied on a specific code list
```

Comment:
Can we really return _all_ in the response? If we can return _all_ then I guess the code field below will be omitted?

Reply:
That is correct. When "all" is used, the code field is never present.